scipy/optimize/tests/test_bracket.py

```diff
@@ -249,13 +250,13 @@ def test_input_validation(self, xp):
         with pytest.raises(ValueError, match=message):
             _bracket_root(lambda x: x, -4+1j, 4)
         with pytest.raises(ValueError, match=message):
-            _bracket_root(lambda x: x, -4, 'hello')
```

Comment:
It was a bit overkill to expect the same message for extremely odd input like this. Now we call `array_namespace` on all arguments like we should, so when `SCIPY_ARRAY_API=1`, we get the standard message that only numeric input is accepted. When `SCIPY_ARRAY_API` is not set, we get the same message as we always have, but it's annoying to change the test depending on the value of `SCIPY_ARRAY_API`, so I just test for complex input now.